internal/pod/imagepuller.go

```diff
@@ -43,9 +58,9 @@ func NewImagePuller(client client.Client, scheme *runtime.Scheme) ImagePuller {
 func (ipi *imagePullerImpl) CreatePullPod(ctx context.Context, imageSpec *kmmv1beta1.ModuleImageSpec,
 	micObj *kmmv1beta1.ModuleImagesConfig) error {
 
-	restartPolicy := v1.RestartPolicyOnFailure
+	pullPodTypeLabeValue := pullPodUntilSuccess
 	if imageSpec.Build != nil || imageSpec.Sign != nil {
-		restartPolicy = v1.RestartPolicyNever
+		pullPodTypeLabeValue = pullPodTypeOneTime
```

Comment:
Shouldn't it be the opposite? That we run once if there are not build/sign and continue trying if we can build?

Reply:
actually  - no. In case there is a buld/sign present, then once the pull fails, we can stop trying to pull and go to building/signing images. In case there are no build/sign, then we need to try the pulling indefinitly

Reply:
Ohh right, and then we update the MIC status based on the MBSC status and not based on the pull pods.